### PR TITLE
Feat/referral

### PR DESCRIPTION
**Pull request by jjBlockchain**
## Description (optional)
* add referral to signup form.

### CODE REVIEW

File: packages/blockchain-wallet-v4-frontend/src/data/signup/sagas.ts [hunk 1]
```diff
@@ -64,11 +64,26 @@ export default ({ api, coreSagas, networks }) => {
       yield put(actions.logs.logErrorMessage(logLocation, 'exchangeMobileAppSignup', e))
     }
   }
+
+  const checkReferralCode = function* (referral) {
+    try {
+      yield call(api.checkIsValidReferralCode, referral)
```
Comment (milan-bc) [inline packages/blockchain-wallet-v4-frontend/src/data/signup/sagas.ts:70]: you do not need to get anything from response and use it?

File: packages/blockchain-wallet-v4/src/redux/wallet/sagas.ts [hunk 2]
```diff
@@ -103,6 +104,10 @@ export default ({ api, networks }) => {
       networks.btc
     )
     yield call(api.createWallet, email, captchaToken, wrapper, forceVerifyEmail)
+    // insert api to set referral code
+    if (referral) {
+      // yield call()
+    }
```
Comment (milan-bc) [inline packages/blockchain-wallet-v4/src/redux/wallet/sagas.ts:110]: is this by intention? if we do not need it maybe to remove?

Comment (jjBlockchain) [reply packages/blockchain-wallet-v4/src/redux/wallet/sagas.ts:110]: oops good 👀

File: packages/blockchain-wallet-v4-frontend/src/data/signup/sagas.ts (hunk 1 above)
Comment (milan-bc) [reply packages/blockchain-wallet-v4-frontend/src/data/signup/sagas.ts:70]: cool, I thought so 👍

**Review by pedroapfilho — APPROVED** [review]

**Pull request merged**